File: src/main/java/com/github/njuro/jard/ban/UserBannedException.java
package com.github.njuro.jard.ban;

import java.io.Serial;
import lombok.NoArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ResponseStatus;

/** Exception to be thrown when banned user attempts to post. */
@ResponseStatus(code = HttpStatus.FORBIDDEN)
@NoArgsConstructor
public class UserBannedException extends RuntimeException {
  @Serial private static final long serialVersionUID = 8772236780722701190L;

  public UserBannedException(String message) {
    super(message);
  }
}

File: src/main/kotlin/com/github/njuro/jard/rewrite/ban/UserBannedException.kt
package com.github.njuro.jard.rewrite.ban

import org.springframework.http.HttpStatus.FORBIDDEN
import org.springframework.web.bind.annotation.ResponseStatus

/** Exception to be thrown when banned user attempts to post.  */
@ResponseStatus(code = FORBIDDEN)
class UserBannedException(message: String) : RuntimeException(message)
